### ISSUE TITLE
ISO 32001 developer extensions dictionary needed for use of a signing certificate signed with ecdsa-with-SHA3-256?

### ISSUE THREAD

**Issue by a20dev** [pdf-association/pdf-issues#241]
Suppose a PDF 2.0 document is to be signed using SHA-256 (not SHA3!) for computing the hash over the document. The signing certificate is signed by its issuer using algorithm ecdsa-with-SHA3-256 (here we have SHA3). Shall the ISO 32001 developer extensions dictionary be added in that case?

The issuer's certificate isn't directly involved in signing the document (as per 5.1.1 of ISO 32001), but it might be embedded in the signature (as intermediate certificate in Contents).

Anyway, software validating the signature has to support SHA3 in this case for validating the certificate chain (just as it has to support SHA3 for documents hashed with SHA3). This makes a point for requiring adding the ISO 32001 developer extensions dictionary in the case being presented here.

**Comment by MatthiasValvekens** [comment]
I don't remember discussing this specific instance of it, but we did discuss a related question in the Encryption & DigSig discussion group at ISO, some time last year: does ISO 32002 support entail being able to process _certificates_ signed with Ed25519?

I don't recall the finer details, but the general consensus was that that was out of scope. Here are some reasons (not all of which were discussed in that meeting, FWIW):

 - Many PDF implementations delegate certificate chain-of-trust validation to external tools, and those external libraries could very well support a different set of cryptographic primitives than the PDF processor itself.
 - Moreover, ISO 32000-x is completely silent on the issue of certificate validation (with the possible exception of the clause on seed values, but that's a different story). It makes sense to keep it that way.
 - The chain of trust is in itself independent of the PDF and not something controlled by the author or even by the issuer of their certificate! In principle, a certificate/signature/public key can have any number of chains of trust, but putting those together and choosing an appropriate one is the validator's responsibility. In other words, the signer is strictly speaking not in any position to make claims about which algorithms the validator has to support in order for trust validation to succeed. (Granted, this objection is probably academic in 99% of actual document signing use cases, but I don't think it can be dismissed offhand.)

Long story short, I believe these arguments still hold water in the situation you describe. As such, I don't think including the ISO 32001 extension would be _mandatory_ per se. But I think an argument can be made that including it anyway is beneficial in many situations. That certainly wouldn't violate any of the provisions in the extension documents as far as I'm aware :).

**Comment by a20dev** [comment]
Thanks!

This also applies to OCSP responses in the DSS which are even more out of scope for ISO 3200[012].

**Comment by petervwyatt** [comment]
PDF TWG agree no fix.
